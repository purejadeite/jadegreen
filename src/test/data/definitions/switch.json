{
	"id": "book",
	"sheets": [
		{
			"comment": "単一シート1件読み込み",
			"id": "main",
			"output": true,
			"target": {
			    "name": "bean"
			},
		    "cells": [
		        {"id": "lang", "value": "ja"},
		        {"id": "logicalClassName", "row": 1, "column": 3},
		        {"id": "className", "row": 2, "column": 3},
		        {"id": "package", "row": 3, "column": 3},
		        {"id": "imports", "row": 4, "column": 3, "splitter": "\n", "options": [
		            {"type": "switch", "cellId": "type", "cases": [
		                {"value": "List<LineBean>", "options": [
		                    {"type": "addvalue", "value": "java.util.ArrayList", "unique": true},
		                    {"type": "addvalue", "value": "java.util.List", "unique": true},
		                    {"type": "addvalue", "value": "com.purejadeite.generated.LineBean", "unique": true}
		                ]},
		                {"value": "Map<String, LineBean>", "options": [
		                    {"type": "addvalue", "value": "java.util.HashMap", "unique": true},
		                    {"type": "addvalue", "value": "java.util.Map", "unique": true},
		                    {"type": "addvalue", "value": "com.purejadeite.generated.LineBean", "unique": true}
		                ]}
		            ]}
		        ]},
		        {"id": "fields", "row": 8, "breakId": "fieldName", "options": [{"type": "sort", "keyId": "fieldName"}],
		            "columns": [
        		        {"id": "type", "value": "line"},
		                {"id": "logicalFieldName", "column": 2},
		                {"id": "fieldName", "column": 3},
		                {"id": "type", "column": 4},
		                {"id": "initial", "column": 5}
		            ]
		        }
		    ]
		}
	]
}