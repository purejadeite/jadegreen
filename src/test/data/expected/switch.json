[
    {
        "className": "Sample",
        "imports": [
            "java.io.Serializable",
            "java.util.ArrayList",
            "java.util.List",
            "com.purejadeite.generated.LineBean",
            "java.util.HashMap",
            "java.util.Map"
        ],
        "lang": "ja",
        "logicalClassName": "単一定義・単一コンテンツ",
        "package": "com.purejadeite.generated",
        "fields": [
            {
                "fieldName": "fileExistFlag",
                "initial": "Boolean.TRUE",
                "logicalFieldName": "ファイル有無フラグ",
                "type": "Boolean"
            },
            {
                "fieldName": "fileName",
                "initial": "null",
                "logicalFieldName": "ファイル名",
                "type": "String"
            },
            {
                "fieldName": "fileNameJp",
                "initial": "null",
                "logicalFieldName": "ファイル名日本語",
                "type": "String"
            },
            {
                "fieldName": "filePath",
                "initial": null,
                "logicalFieldName": "ファイルパス",
                "type": "String"
            },
            {
                "fieldName": "lines",
                "initial": "new ArrayList<>()",
                "logicalFieldName": "行情報",
                "type": "List<LineBean>"
            },
            {
                "fieldName": "map",
                "initial": "new HashMap<>()",
                "logicalFieldName": "行情報マップ",
                "type": "Map<String, LineBean>"
            }
        ]
    }
]